### PR TITLE
Hide reports on closed orders greater than threshold time

### PR DESCRIPTION
**Pull request by aanil**

### CODE REVIEW

File: daily_read/order_portal.py [hunk 1]
```diff
@@ -73,14 +73,17 @@ def process_orders(self, priority, closed_before_in_days=30):
...
-                continue
-            elif order["identifier"] not in self.projects_data.data.keys():
+            delete_report = False
+            if order["identifier"] not in self.projects_data.data.keys():
                 log.debug(f"Order portal id: {order['identifier']} not found in data fetched from sources")
                 continue
+            elif order["status"] == "closed":
+                # Add 2 day padding for deleting reports from closed projects
+                close_date = datetime.datetime.strptime(order["history"]["closed"], "%Y-%m-%d").date()
+                if close_date < older_than_cutoff - datetime.timedelta(days=2):
```
Comment (alneberg) [inline daily_read/order_portal.py]: Could you add some more comments here to help the reader figuring out what's going on?

Comment (alneberg) [reply daily_read/order_portal.py]: That's for this section, not the specific row necessarily

Comment (aanil) [reply daily_read/order_portal.py]: Added some more comments now!

File: daily_read/order_portal.py [hunk 2]
```diff
@@ -100,43 +103,52 @@ def process_orders(self, priority, closed_before_in_days=30):
...
+                order_updates_item["active_projects"] += 1
 
         return order_updates
 
-    def upload_report_to_order_portal(self, report, project):
-        """Upload report to order portal"""
+    def upload_report_to_order_portal(self, report, project, status):
+        """Upload report to order portal
+        With the status 'published' the user can see the report immediately
+        With the status 'review', the user will not be able to view the report(used here as a proxy for deletion)
```
Comment (alneberg) [inline daily_read/order_portal.py:137]: So I guess it's not possible to delete a report then?

File: daily_read/order_portal.py (hunk 1 above)
Comment (alneberg) [reply daily_read/order_portal.py]: Thank you!

File: daily_read/order_portal.py (hunk 2 above)
Comment (aanil) [reply daily_read/order_portal.py:137]: Reports can't be deleted through the API currently, but only through the UI. We could ask for the functionality to be added, I guess

**Comment by alneberg** [comment]
I wonder if we should increase the padding to more days? Is it OK to set a report to review multiple times?

File: daily_read/order_portal.py (hunk 2 above)
Comment (alneberg) [reply daily_read/order_portal.py:137]: Yes sounds like a good thing to ask for

**Comment by aanil** [comment]
> I wonder if we should increase the padding to more days? Is it OK to set a report to review multiple times?

Yeah, it seems like you can call the API with the same status, it will stay the same.

**Comment by alneberg** [comment]
Ok, let's make it 5 days then. I'm a bit worried that if LIMS2DB or Statusdb would be down, there would be some reports not deleted.

**Comment by aanil** [comment]
> Ok, let's make it 5 days then. I'm a bit worried that if LIMS2DB or Statusdb would be down, there would be some reports not deleted.

Done!

**Review by alneberg — APPROVED** [review]
Great work!

**Pull request merged**